providers/implementations/kdfs/hkdf.c.in

```diff
@@ -966,61 +1003,77 @@ static int kdf_tls1_3_derive(void *vctx, unsigned char *key, size_t keylen,
     }
 }
 
+#define kdf_tls1_3_set_ctx_params_st    hkdf_all_set_ctx_params_st
+
+{- produce_param_decoder('kdf_tls1_3_set_ctx_params',
+                         (['KDF_PARAM_MODE',              'mode',   'utf8_string'],
+                          ['KDF_PARAM_MODE',              'mode',   'int'],
+                          ['KDF_PARAM_PROPERTIES',        'propq',  'utf8_string'],
+                          ['ALG_PARAM_ENGINE',            'engine', 'utf8_string', 'hidden'],
+                          ['KDF_PARAM_DIGEST',            'digest', 'utf8_string'],
+                          ['KDF_PARAM_KEY',               'key',    'octet_string'],
+                          ['KDF_PARAM_SALT',              'salt',   'octet_string'],
+                          ['KDF_PARAM_FIPS_KEY_CHECK',    'ind_k',  'int'],
```

Comment:
I would prefer if the fips indicators were not abbreviated..
since we have to map SETTABLE indexes to these values..
the name should be something that maps nicely to FIPS_KEY_CHECK..

We have bits of code that do this....

````
        if (!OSSL_FIPS_IND_ON_UNAPPROVED(ctx, OSSL_FIPS_IND_SETTABLE0,
                                         libctx, "HKDF", "Key size",
                                         ossl_fips_config_hkdf_key_check)) {
            ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_KEY_LENGTH);
````
and
````
    if (!OSSL_FIPS_IND_SET_CTX_PARAM(ctx, OSSL_FIPS_IND_SETTABLE0, params,
                                     OSSL_KDF_PARAM_FIPS_KEY_CHECK))
````